patterns/1-initial/internal-developer-platform.md

```diff
@@ -0,0 +1,145 @@
+
+**Increasing employee happiness**: The number one objective of an IDP is to increase Developer happiness, by treating and promoting the platform as an internal product. By reducing friction in the development process, an IDP allows developers to focus on what they do best which is writing code and solving meaningful problems. Self-service capabilities reduce waiting times for resources, while centralized platforms eliminate the complexity of managing disparate tools. This leads to a more streamlined developer experience, which in turn boosts job satisfaction and productivity​.
+
+**Increasing success of new hires**: IDPs simplify the onboarding process by providing new hires with easy access to pre-configured environments, documentation, and templates. Rather than spending their first weeks figuring out how to set up their development environment or learning which tools to use, new hires can start contributing much faster. The IDP serves as a single point of entry for everything they need, dramatically reducing ramp-up time​. As a result, developer onboarding time (typically measured through time for the first approved commit) is a key metric in IDP implementation and success.
+
+**Building actionable documentation**: An IDP integrates documentation directly into the development workflows via self-service portals. Teams can access API documentation, code examples, and templates from the same place they access tools and environments. This centralization ensures that documentation is kept up-to-date and easily accessible. Furthermore, by encouraging direct InnerSource contributions into the documentation itself (from users), teams can continuously improve documentation based on real-world usage​.
+
+## Rationale
+
+An IDP provides a solution to the competing forces of consistency, flexibility, and scalability. By centralizing the management of development environments and tooling while offering self-service capabilities, it balances the needs of both developers and platform teams. InnerSource practices further contribute by fostering a culture of collaboration and reuse across the organization​. It supports scalable and controlled growth within the internal software development community, facilitating the adoption of wider InnerSource practices.
+
+## Known Instances (optional)
+
+(known instances to be added)
+
+## Status
+
+Initial Draft
+
+## Author(s)
+
+Vincent Caldeira
+
+## Acknowledgments (optional)
+
+Include those who assisted in helping with this pattern - both for attribution and for possible future follow up.
+Though optional, most patterns should list who helped in their creation.
+
```

Comment:
It could be good add a `References` chapter to link this pattern with others that could be related, such as:

* [InnerSource Portal](https://patterns.innersourcecommons.org/p/innersource-portal)

Maybe others could be listed here too. Ideas?